evaluation/math_test.go

```diff
@@ -0,0 +1,219 @@
+package evaluation
+
+import (
+	"../engines"
+	"../structs"
+	"testing"
+)
+
+func TestPlus(t *testing.T) {
+	types, functions, bindings := structs.Maps()
+
+	expressions, _ := engines.Translate("(+ 1 2)")
+	test1, _ := plus(expressions, &types, &functions, &bindings)
+
+	expressions, _ = engines.Translate("(+ 1 2 3)")
+	test2, _ := plus(expressions, &types, &functions, &bindings)
+
+	expressions, _ = engines.Translate("(+ 1 -4)")
+	test3, _ := plus(expressions, &types, &functions, &bindings)
+
+	expressions, _ = engines.Translate("(+ 1 (+ 1 2))")
+	test4, _ := plus(expressions, &types, &functions, &bindings)
+
+	expressions, _ = engines.Translate("(+ 1 (+ (+ 1 2) 1))")
+	test5, _ := plus(expressions, &types, &functions, &bindings)
+
+	expressions, _ = engines.Translate("(+ 1)")
+	test6, _ := plus(expressions, &types, &functions, &bindings)
+
+	expressions, _ = engines.Translate("(+)")
+	_, err7 := plus(expressions, &types, &functions, &bindings)
+
+	expressions, _ = engines.Translate("(+ $ #)")
+	_, err8 := plus(expressions, &types, &functions, &bindings)
+
+	if test1 != 3 {
+		t.Errorf("(+ 1 2) = %v; want 3", test1)
+	}
+	if test2 != 6 {
+		t.Errorf("(+ 1 2 3) = %v; want 6", test2)
+	}
+	if test3 != -3 {
+		t.Errorf("(+ 1 -4) = %v; want -3", test3)
+	}
+	if test4 != 4 {
+		t.Errorf("(+ 1 (+ 1 2)) = %v; want 4", test4)
+	}
+	if test5 != 5 {
+		t.Errorf("(+ 1 (+ (+ 1 2) 1)) = %v; want 5", test5)
+	}
+	if test6 != 1 {
+		t.Errorf("(+ 1) = %v; want 1", test6)
+	}
+	if err7 == nil {
+		t.Errorf("(+) = %v; want invalid number of arguments.", err7)
+	}
+	if err8 == nil {
+		t.Errorf("(+ $ #) = %v; want invalid syntax.", err8)
+	}
+}
```

Comment:
suggested addition: (+ '(1 2 3) '(4 5 6)) should say it doesn't work on lists or something like that, same with adding bools; do the same thing for all arithmetic operations